flink-runtime/src/test/java/org/apache/flink/runtime/io/network/NettyShuffleEnvironmentCleanupTest.java

```diff
@@ -0,0 +1,181 @@
+		config.setString(JobManagerOptions.ADDRESS, "localhost");
+		config.setString(RestOptions.BIND_PORT, "0");
+		config.setString(HighAvailabilityOptions.HA_MODE, "zookeeper");
+		config.setString(HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM, zooKeeperResource.getConnectString());
+		config.setString(HighAvailabilityOptions.HA_STORAGE_PATH, temporaryFolder.newFolder().getAbsolutePath());
+		config.setString(CoreOptions.TMP_DIRS, taskManagerTmpDir.getAbsolutePath());
+		config.setString(TaskManagerOptions.MANAGED_MEMORY_SIZE, "4m");
+		config.setInteger(NettyShuffleEnvironmentOptions.NETWORK_NUM_BUFFERS, 100);
+		config.setString(TaskManagerOptions.TOTAL_FLINK_MEMORY, "512m");
+
+		try (FileOutputStream fos = new FileOutputStream(confFile);
+			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(fos))) {
+			for (Map.Entry<String, String> e : config.toMap().entrySet()) {
+				writer.write(e.getKey());
+				writer.write(": ");
+				writer.write(e.getValue());
+				writer.newLine();
+			}
+
+			writer.flush();
+		}
+
+		TaskManagerProcess taskManagerProcess = null;
+
+		try (final StandaloneSessionClusterEntrypoint clusterEntrypoint = new StandaloneSessionClusterEntrypoint(config)) {
+			String javaCommand = getJavaCommandPath();
+			if (javaCommand == null) {
+				fail("Could not find java executable.");
+			}
+
+			clusterEntrypoint.startCluster();
+
+			taskManagerProcess = new TaskManagerProcess(javaCommand, confDir.getAbsolutePath());
+			taskManagerProcess.startProcess();
+
+			// Waits till netty shuffle environment has created the tmp directories.
+			Deadline deadline = Deadline.now().plus(TEST_TIMEOUT);
+			File[] nettyShuffleTmpFiles = findNettyShuffleDirectories(taskManagerTmpDir);
+
+			while (nettyShuffleTmpFiles.length == 0 && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+				nettyShuffleTmpFiles = findNettyShuffleDirectories(taskManagerTmpDir);
+			}
+
+			if (nettyShuffleTmpFiles.length == 0) {
+				fail("The TaskManager process does not create shuffle directories in time, " +
+						"its output is: \n" + taskManagerProcess.getProcessOutput());
+			}
+
+			Process kill = Runtime.getRuntime().exec("kill " + taskManagerProcess.getProcessId());
+			kill.waitFor();
+			assertEquals("Failed to send SIG_TERM to process", 0, kill.exitValue());
+
+			deadline = Deadline.now().plus(TEST_TIMEOUT);
+			while (taskManagerProcess.isAlive() && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+			}
+
+			if (taskManagerProcess.isAlive()) {
+				fail("The TaskManager process does not terminate in time, its output is: \n" + taskManagerProcess.getProcessOutput());
+			}
+
+			// Checks if the directories are cleared.
+			nettyShuffleTmpFiles = findNettyShuffleDirectories(taskManagerTmpDir);
+			assertEquals("The TaskManager does not remove the tmp shuffle directories after termination, " +
+							"its output is \n" + taskManagerProcess.getProcessOutput(),
+					0, nettyShuffleTmpFiles.length);
+		}
```

Comment:
we should add a finally part to cleanup the `FileChannelManagerImpl` as the bottom line. In case this test fails unexpected, we still do not leak any files.

Reply:
Agree with that we should add the finally block, and destroy the process in the finally block. Besides, I think we should not need to delete files explicitly since we are using the `TemporaryFolder` and it should be able to ensure files get cleared after the test.